fieldsolver/derivatives.cpp

```diff
@@ -775,7 +783,7 @@ void calculateScaledDeltas(
    cell->parameters[CellParams::AMR_DBSQ] = dBsq;
    cell->parameters[CellParams::AMR_DB] = dB;
    cell->parameters[CellParams::AMR_ALPHA] = alpha;
-   cell->parameters[CellParams::AMR_JPERB] = J / Bperp;
+   cell->parameters[CellParams::AMR_JPERB] = J / (Bperp + EPS);   // Epsilon in denominator so we don't get infinities
```

Comment:
Is that Arto's constant from somewhere, like 1e-30? Wouldn't we be safer to set the whole ratio to zero if Bperp < EPS, instead of every now and then risking a spike if Bperp is small?

Reply:
I don't think a spike here is necessarily an issue, it will just end up as that cell wanting as high refinement as possible. Maybe the epsilon could be adjusted? MHD-AEPIC uses 9 nT.

Reply:
Well OK I suppose spots with little field do indicate a current sheet or reconnection.